internal/transport/transporter.go

```diff
@@ -10,7 +14,7 @@ type DataReceiveHandlerFunc func([]byte, string)
 type Transporter interface {
 	Connect() error
 	Disconnect(quiesce uint)
-	SendData(data []byte, dest string) error
+	SendData(data []byte, dest string) (*yggdrasil.TransportResponse, error)
```

Comment:
The thing is that if we do in this way, we lost the `status_code` (And we need it), I'm ok to have another struct here similar to this. I followed the same as yggdrasil.APIResponseError in this case. 

Are you ok, if I set transport.APIResponse similar to yggdrasil one?

Reply:
What about: `SendData(data []byte, dest string) (code int, resp []byte, err error)`?

Or maybe even define `code` as an enum?

```
type TransporterResponseCode int

const (
    TransporterResponseCodeSuccess TransporterResponseCode = 0
    TransporterResponseCodeFail    TransporterResponseCode = 1
)

/* ... */

SendData(data []byte, dest string) (code TransporterResponseCode, resp []byte, err error)
```

Reply:
sorry! also the header can be needed, just because of any Aut{n,z} token inside the headers :-( Thinking on openid-connect.

Reply:
so, SendData is a json.Marshal(*yggdrasil.TransportResponse) and return bytes, is that what you suggested? I'm happy to do that, but I do not think that makes sense, workers need to know what kind of transport the endpoint is using if wants to get the SendData response. 

But solution works for us.

Reply:
Well, we've created the `transport.Transporter` interface to abstract away the details of the actual network protocol being used to transport the data. The current interface has two reflective methods in its definition: `SendData` and `ReceiveData`. Because the nature of the underlying network protocol can vary, we kept this interface very simple: "send some bytes" and "get some bytes". As you know, HTTP is a synchronous request/response protocol, while MQTT is an asynchronous publish/subscribe protocol. What we need to find is some sort of common ground here that works elegantly as an extension of the `Transporter` interface that both MQTT and HTTP as implementations of this interface can easily support.

The objective here is to extend the tx/rx nature of the `Transporter` interface to include the concept of a tx-response (and theoretically an rx-response). These responses should be common between all implementations (currently MQTT and HTTP). Does it matter to the `Transporter` interface what these responses look like? Does it matter to the caller of the interface methods what these responses look like? If the "response" value is not opaque, what content is meaningful to a caller of the `Transporter` interface methods? What sort of data would one expect back from a `SendData` or a `ReceiveData` call?

This is only half the objective. The other half is expanding the RPC protocol to relay these transport responses, but that should be considered in its own right. I would consider similar questions: what does a response to a "Send" RPC method look like? What sort of data is meaningful to the caller of the "Send" RPC method?

Reply:
Ok, got it. I'll submit a few changes early next week.